### ISSUE TITLE
Inconsistent last_update

### ISSUE THREAD

**Issue by horstle** [emilsvennesson/script.module.inputstreamhelper#344]
I think we should have another thought about `last_update`, which iirc should more appropriately be called `last_checked` or something. Maybe we could also add a real `last_updated`, which would reflect when a new version was installed. Not sure whether that is needed and how to handle backups/rollbacks though.

**Comment by mediaminister** [comment]
I think we should keep this as simple as possible. We only need the timestamp of the previous update check. This can be called `prev_update_check`. 

FYI, the current implementation doesn't work as expected, I got the following remark from a user a couple of weeks ago:

> I had an update of Widevine yesterday, and it occurred to me that I had to approve the update quite a few times (3 or 4 times if I remember correctly). One messages just showed that I had to have a few command line tools, but I assume it can be detected fine if I have these on my system so there is no need to show this message.
> 6:28
> I think two messages would be fine. One to approve the update, and one to accept the EULA.

**Comment by dagwieers** [comment]
@horstle This is what I would prefer. I'd like to know when it was last checked, and I would like to know when it was last updated.

If we roll back, we should update last_updated as well IMO.

**Comment by dagwieers** [comment]
Let's fix this for v0.5.0 as well.

**Comment by horstle** [comment]
One part of this is pretty much done. Now we have to decide what to do about the remaining usage of last_update.

Some ideas:
* Remove it. Instead, show last check in info dialog.
* Keep it tied to the widevine version (i.e. when that version was installed), e.g. in the manifest.json. In this case, should it be changed when the same version is installed again manually?
* Set to current time whenever a different version is installed, whether it's a new one or from backup.

**Comment by dagwieers** [comment]
Thanks for this!

I would use the current time every time a change is made. This reflects when it was updated, and not "what was updated" or "when it was downloaded first".

**Comment by horstle** [comment]
So any time the content of the cdm directory is changed, the time should be updated?
The only functions that change it are `install_cdm_from_backup` and `remove_widevine`, so we'd only need to set that setting there.

**Comment by dagwieers** [comment]
Everytime the CDM is changed, that value should be update IMO.
We could call `last_update` → `last_changed`, and maybe call `last_check` → `last_checked`.

What do others think?

**Comment by mediaminister** [comment]
The information displayed in the GUI must be unambiguous to the user. Currently, this is not the case.

I think the GUI should show this:
- Widevine not found --> don't display a date
- Widevine version found --> display installation date

> Set to current time whenever a different version is installed, whether it's a new one or from backup.
I think that is the most logical solution, this means that `last_check` is not shown in the GUI anymore and instead `last_update` is shown.

**Comment by dagwieers** [comment]
I think there is a value in showing when it was last checked as well, so people can confirm the check has happened, but e.g. failed to find an update, or failed to offer the update. If people copy&paste or make a screenshot we can determine it has performed the check. (So ideally the update should happen just after a successful check...)

**Comment by mediaminister** [comment]
> I think there is a value in showing when it was last checked as well

I understand, but this means adding another line to the already crowded InputStreamHelper Information dialog.

**Comment by horstle** [comment]
> I think the GUI should show this:
> 
>     * Widevine not found --> don't display a date
> 
>     * Widevine version found --> display installation date

> I think that is the most logical solution, this means that `last_check` is not shown in the GUI anymore and instead `last_update` is shown.

That's how it would be after #360 is merged. The only thing missing is that `last_update` (or whatever we'll call it) would also have to be updated when widevine is installed from a backup and would have to be removed when widevine is removed. I'll add another commit for that in ~1/2h.

**Comment by dagwieers** [comment]
Agreed, I am not a big fan of the *textviewer* widget myself :-)
We show 3 titles, maybe that can be improved as well?

We could also provide more condensed information, how about:

> **Kodi** is running version **18.6** on platform **Linux** and architecture **arm**.
> **InputStream Helper is at version **0.4.7**
> **InputStream Adaptive as at version **2.4.5.1**
> **Widevine CDM** has version **4.10.1610.6** and was last updated on **2020-05-07 13:57**
> **Widevine CDM** was last checked for updates on **2020-05-29 12:03**
> **Widevine CDM** is stored in **/storage/.kodi/cdm** 
> **Widevine CDM** was extracted from ChromeOS **12739.111.1** from image **BIG**
>
> Please report issues to: **github.com/emilsvennesson/script.module.inputstreamhelper/issues**

**Comment by horstle** [comment]
I think we talked about using an url shortener previously. Might be a good idea. https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/ would be https://git.io/JfKJb

**Comment by horstle** [comment]
I changed `last_update` to `last_modified` and moved the `set_setting()` as discussed. I think the changes to info dialog should be in another PR.

**Comment by dagwieers** [comment]
Very well. Thanks for this!

**Comment by dagwieers** [comment]
We can close this now.
